### ISSUE TITLE
node version 9 installed for 8.9.0

### ISSUE THREAD

**Issue by sudo-suhas** [coreybutler/nvm-windows#317]
### My Environment
Windows 10

### I'm using NVM4W version:
1.1.6
### I have already...

- [x] read the README to be aware of npm gotchas & antivirus issues.
- [x] reviewed the [wiki](https://github.com/coreybutler/nvm-windows/wiki) to make sure my issue hasn't already been resolved.
- [x] verified I'm using an account with administrative privileges.
- [x] searched the issues (open and closed) to make sure this isn't a duplicate.
- [x] made sure this isn't a question about how to use NVM for Windows, since [gitter](https://gitter.im/coreybutler/nvm-windows?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) is used for questions and comments.

### My issue is related to (check only those which apply):

- [ ] settings.txt
- [ ] proxy support (Have you tried version 1.1.0+?)
- [ ] 32 or 64 bit support (Have you tried version 1.1.3+?)
- [ ] Character escaping (Have you tried version 1.1.6+?)
- [ ] A standard shell environment (terminal/powershell)
- [ ] A non-standard shell environment (Cmder, Hyper, Cygwin, git)

### Expected Behavior

```
λ nvm use 8.9.0
Now using node v8.9.0 (64-bit)

λ node --version
v8.9.0
```

### Actual Behavior

```
λ nvm use 8.9.0
Now using node v8.9.0 (64-bit)

λ node --version
v9.0.0
```

### Steps to reproduce the problem:

```
λ nvm install 8.9.0
...

λ nvm use 8.9.0
Now using node v8.9.0 (64-bit)

λ node --version
```

Once I installed `8.9.1`, the issue was fixed. I apologise if this is a bug upstream.

**Comment by sudo-suhas** [comment]
I am very confused as to why and how but:

```
λ nvm use 4.8.4
Now using node v4.8.4 (64-bit)

λ node --version
v8.4.0
```

Not sure if this is relevant but I had tried to delete previous version of node using nvm which I was no longer using. I was forced to delete folders manually after `nvm uninstall` failed.

I completely uninstalled `nvm` and reinstalled it. Now things seem to be working fine.

**Comment by sudo-suhas** [comment]
I think this happens because of [`choco install yarn`](https://github.com/yarnpkg/yarn/issues/2524). Furthermore, node version will keep getting overwritten during `choco upgrade all`(upgrades installed packages and their dependencies). Maybe add a note to readme warning about this? The issue can be fixed by doing the following:
```
C:\Windows\system32>choco uninstall yarn -y --remove-dependencies
Chocolatey v0.10.8
Uninstalling the following packages:
yarn

yarn v1.3.2
 Running auto uninstaller...
 Auto uninstaller has successfully uninstalled yarn or detected previous uninstall.
 yarn has been successfully uninstalled.

nodejs.install v9.1.0
 Running auto uninstaller...
 Auto uninstaller has successfully uninstalled nodejs.install or detected previous uninstall.
 nodejs.install has been successfully uninstalled.
Environment Vars (like PATH) have changed. Close/reopen your shell to
 see the changes (or in powershell/cmd.exe just type `refreshenv`).

Chocolatey uninstalled 2/2 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

C:\Windows\system32>choco install yarn --ignore-dependencies
Chocolatey v0.10.8
Installing the following packages:
yarn
By installing you accept licenses for the packages.
Progress: Downloading yarn 1.3.2... 100%

yarn v1.3.2 [Approved]
yarn package files install completed. Performing other installation steps.
Downloading yarn
  from 'https://yarnpkg.com/downloads/1.3.2/yarn-1.3.2.msi'
Progress: 100% - Completed download of C:\Users\suhas.karanth\AppData\Local\Temp\chocolatey\yarn\1.3.2\yarn-1.3.2.msi (1.24 MB).
Download of yarn-1.3.2.msi (1.24 MB) completed.
Hashes match.
Installing yarn...
yarn has been installed.
...
  yarn may be able to be automatically uninstalled.
Environment Vars (like PATH) have changed. Close/reopen your shell to
 see the changes (or in powershell/cmd.exe just type `refreshenv`).
 The install of yarn was successful.
  Software installed as 'msi', install location is likely default.

Chocolatey installed 1/1 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
```

**Comment by jacobq** [comment]
Weird. It's like the first digit of the version number was getting ignored / chopped-off. I have never encountered this. Is anyone still having this problem?

**Comment by coreybutler** [comment]
These issues were likely caused by errors in the underlying semver library. They should be resolved in 1.1.7+
